addons/account/wizard/account_move_reversal.py

```diff
@@ -31,22 +31,23 @@ class AccountMoveReversal(models.TransientModel):
     def default_get(self, fields):
         res = super(AccountMoveReversal, self).default_get(fields)
         move_ids = self.env['account.move'].browse(self.env.context['active_ids']) if self.env.context.get('active_model') == 'account.move' else self.env['account.move']
+        if any(move.state != "posted" for move in move_ids):
+            raise UserError(_('You can only reverse posted moves.'))
+        res['move_ids'] = [(6, 0, move_ids.ids)]
         res['refund_method'] = (len(move_ids) > 1 or move_ids.type == 'entry') and 'cancel' or 'refund'
-        res['residual'] = len(move_ids) == 1 and move_ids.amount_residual or 0
-        res['currency_id'] = len(move_ids.currency_id) == 1 and move_ids.currency_id.id or False
-        res['move_type'] = len(move_ids) == 1 and move_ids.type or False
         return res
 
-    @api.depends('move_id')
+    @api.depends('move_ids')
     def _compute_from_moves(self):
-        move_ids = self.env['account.move'].browse(self.env.context['active_ids']) if self.env.context.get('active_model') == 'account.move' else self.move_id
         for record in self:
+            move_ids = record.move_ids
             record.residual = len(move_ids) == 1 and move_ids.amount_residual or 0
             record.currency_id = len(move_ids.currency_id) == 1 and move_ids.currency_id or False
-            record.move_type = len(move_ids) == 1 and move_ids.type or False
+            record.move_type = move_ids.type if len(move_ids) == 1 else (any(move.is_invoice() for move in move_ids) and 'some_invoice' or False)
```

Comment:
is_invoice() will return True if a record's type is 'out_refund' or 'in_refund', which isn't what you want here I think